### ISSUE TITLE
Codeunit 134 Import Attachment - Inc. Doc. Function ImportAttachment External

### ISSUE THREAD

**Issue by ClintonSaundersSeeSide** [microsoft/ALAppExtensions#3785]
Please set the External Property to true on the ImportAttachment function in codeunit 134 so that it can be called from extensions. We have developed a drag and drop utility for the Incomming Document Factbox, and need to call this function. We would like our extension to work on the cloud version.

**Comment by AlexanderYakunin** [comment]
This function is not designed for cloud and specifically tagged as 'OnPrem'. I'm afraid you cannot use it.

**Comment by ClintonSaundersSeeSide** [comment]
Thanks for the feedback, however I would like a little more detail on the logic of your decision please. The Incomming Document factbox is available on the cloud version, accepting files that are uploaded using the file dialog functionality in the standard version. All we want to be able to do it to call the above function from our Drag and Drop utility so that files can be uploaded quicker. Its really crazy not to make it available, because if you don't we can simply rewrite the function anyway as part of the extension, and achieve the same thing. That seems like a crazy thing to do though, when the function that is already there can just be made available?

Hoping for a better answer.

**Comment by AlexanderYakunin** [comment]
@JesperSchulz

**Comment by AlexanderYakunin** [comment]
You need to understand that if we expose this procedure as external, we will not be able to rework this code for the modern state of mailing & attachments for the cloud. You will reach you goal but it will block all other partners and us, as we cannot make breaking changes. You can easily copy this function (but please use your prefix for function name) to your solution and use it.

**Comment by ClintonSaundersSeeSide** [comment]
Thank you for the response. Did not realise that it would be a breaking change... this was not explained in the original answer.

I will copy the function and use in the extension then as indicated.

**Comment by darjoo** [comment]
@ClintonSaundersSeeSide as you are able to unblock yourself by copying the function, I will consider this issue closed. If you still have any issues with it, feel free to open another issue. Thanks.
